cross-platform/react-app/src/frontend/Screens/Hub/HubScreen.tsx

```diff
@@ -5,36 +5,42 @@
 import React from "react";
 import { AlertAction, MobileCore } from "@itwin/mobile-sdk-core";
 import { ActionSheetButton, BackButton, useIsMountedRef } from "@itwin/mobile-ui-react";
-import { ITwin } from "@itwin/itwins-client";
 import { BriefcaseConnection, IModelConnection } from "@itwin/core-frontend";
-import { Button, HubStep, i18n, IModelDownloader, IModelInfo, IModelPicker, presentError, Project, ProjectPicker, Screen, SignIn, signOut, useLocalizedString } from "../../Exports";
+import { Button, HubStep, i18n, IModelDownloader, IModelInfo, IModelPicker, presentError, ProjectPicker, Screen, SignIn, signOut, useLocalizedString } from "../../Exports";
 import "./HubScreen.scss";
 
 HubScreen.ACTIVE_PROJECT_INFO = "activeProjectInfo";
 
 /**
- * Gets active project stored in `localStorage`, or undefined if there isn't one
- * @see {@link saveActiveProject}.
- * @returns A {@link Project} object representing the active project, or undefined if there is none.
+ * Gets the active project id stored in `localStorage`, or undefined if there isn't one.
+ * @see {@link saveActiveProjectId}.
+ * @returns The active project id or undefined.
  */
-function getActiveProject() {
-  const projectInfoJson = localStorage.getItem(HubScreen.ACTIVE_PROJECT_INFO);
-  if (projectInfoJson) {
-    const project = JSON.parse(projectInfoJson);
-    if (project.id) {
-      return project as Project;
+function getActiveProjectId() {
+  const storedValue = localStorage.getItem(HubScreen.ACTIVE_PROJECT_INFO);
+  if (!storedValue) return undefined;
+
+  // Previously we stored the whole Project object, now we just store the id.
+  // This code attempts to properly handle the old data.
+  if (storedValue.includes("{")) {
```

Comment:
I'm OK with this as-is, but if you JSON stringified the ID, you could always do the JSON parse here, then if it the result is an object you can check for the id field, and if it's a string you can just return it. If you feel doing it the other way is better, go ahead and do that. If not, simply resolve this comment.

Reply:
Always stringifying and parsing seems like extra work for no reason, so I'd rather leave it as-is.